synapse/handlers/register.py

```diff
@@ -406,6 +412,32 @@ def register_email(self, threepidCreds):
                     403, "Third party identifier is not allowed"
                 )
 
+    @defer.inlineCallbacks
+    def bind_email_to_account(self, user_id, email, validated_at=None):
```

Comment:
unless you have a reason for factoring this out, i feel like it may as well be inlined.